client/components/tinymce/plugins/mentions/mentions.jsx

```diff
@@ -0,0 +1,341 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import ReactDomServer from 'react-dom/server';
+import { connect } from 'react-redux';
+import { includes } from 'lodash';
+import tinymce from 'tinymce/tinymce';
+
+/**
+ * Internal dependencies
+ */
+import SuggestionList from './suggestion-list';
+import EditorMention from './editor-mention';
+import QueryUsersSuggestions from 'components/data/query-users-suggestions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getUserSuggestions } from 'state/users/suggestions/selectors';
+
+/**
+ * Module variables
+ */
+const VK = tinymce.util.VK;
+
+const getMatchingSuggestions = function( suggestions, query ) {
+	if ( ! query ) {
+		return suggestions.slice( 0, 10 );
+	}
+
+	let matchingSuggestions = suggestions;
+
+	matchingSuggestions = suggestions.filter( ( suggestion ) => {
+		// Start of string or preceded by a space.
+		const matcher = new RegExp( '^' + query + '| ' + query, 'ig' );
+		const name = suggestion.name || suggestion.user_login + ' ' + suggestion.display_name;
+
+		return matcher.test( name );
+	} );
+
+	return matchingSuggestions.slice( 0, 10 );
```

Comment:
looks like we're one step away from getting rid of the `let` here. we could even get rid of `matchingSuggestions` I think

```js
return suggestions
	.filter( suggestion => {

	} )
	.slice( 0, 10 );
```